shared/domains/chelonia/chelonia.js

```diff
@@ -344,25 +471,28 @@ export default (sbp('sbp/selectors/register', {
     }
   },
   'chelonia/latestContractState': async function (contractID: string) {
-    const events = await sbp('chelonia/out/eventsSince', contractID, contractID)
-    let state = {}
+    const events = await sbp('chelonia/private/out/eventsSince', contractID, contractID)
+    if (sbp(this.config.stateSelector)[contractID]) {
+      return cloneDeep(sbp(this.config.stateSelector)[contractID])
+    }
+    let state = Object.create(null)
     // fast-path
     try {
       for (const event of events) {
         await sbp('chelonia/private/in/processMessage', GIMessage.deserialize(event), state)
       }
       return state
     } catch (e) {
-      console.warn(`[chelonia] latestContractState(${contractID}): fast-path failed due to ${e.name}: ${e.message}`)
-      state = {}
+      console.warn(`[chelonia] latestContractState(${contractID}): fast-path failed due to ${e.name}: ${e.message}`, e.stack)
+      state = cloneDeep(sbp(this.config.stateSelector)[contractID] || {})
```

Comment:
What's going on here? I don't think this line should've been changed?

Reply:
I think you're right, will fix it..